### ISSUE TITLE
Convert truncates microseconds from OffsetDateTime when converting to Instant

### ISSUE THREAD

**Issue by t9t** [jOOQ/jOOQ#13913]
### Expected behavior

When I call `Record.get(FIELD, Instant.class)` it should return an `Instant` with the same precision as stored in the database.

### Actual behavior
- `Record.get(FIELD, Instant.class)` truncates the result to milliseconds
- `Record.get(FIELD, OffsetDateTime.class)` returns an `OffsetDateTime` with the expected precision (microseconds in my case)

### Steps to reproduce the problem

This is the smallest test case I could come up with, using PostgreSQL (I don't know if it's specific to that):

```
// dsl injected from a Spring context in this case
var record = dsl.selectFrom("(select current_timestamp) c").fetchOne();
System.out.println("record: " + record);
System.out.println("Instant: " + record.get(0, Instant.class));
System.out.println("OffsetDateTime: " + record.get(0, OffsetDateTime.class));
```

Printed output:
```
record: +--------------------------------+
|current_timestamp               |
+--------------------------------+
|2022-08-23T18:58:59.658266+02:00|
+--------------------------------+

Instant: 2022-08-23T16:58:59.658Z
OffsetDateTime: 2022-08-23T18:58:59.658266+02:00
```

### Versions

- jOOQ: 3.17.3
- Java: OpenJDK 64-Bit Server VM Zulu17.34+19-CA (build 17.0.3+7-LTS, mixed mode, sharing)
- Database (include vendor): `postgres:13-alpine` Docker image
- OS: MacOS
- JDBC Driver (include name if unofficial driver): `org.postgresql:postgresql:42.3.6`

Possibly related:
- https://stackoverflow.com/questions/68437631/jooqs-record-mapper-converter-losing-timestamp-precision
- https://github.com/jOOQ/jOOQ/issues/12225

**Comment by lukaseder** [comment]
Thanks for your report. Indeed, that's the same problem as #12225, which wasn't implemented correctly for all pairs of temporal data types. Will look into this soon.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.4 (https://github.com/jOOQ/jOOQ/issues/13923), and 3.16.10 (https://github.com/jOOQ/jOOQ/issues/13924).

Thanks again for your report!

**Comment by t9t** [comment]
Thanks for the quick fix _and_ release of the fix!
